Sources/Controllers/2.0/OARearrangeCustomFiltersViewController.mm

```diff
@@ -0,0 +1,453 @@
+@property (nonatomic) OAPOIUIFilter *filter;
+
+- (instancetype) initWithFilter:(OAPOIUIFilter *)filter;
+
+@end
+
+@implementation OAEditFilterItem
+
+- (instancetype) initWithFilter:(OAPOIUIFilter *)filter
+{
+    self = [super init];
+    if (self) {
+        _filter = filter;
+        _order = filter.order;
+    }
+    return self;
+}
+
+@end
+
+@interface OAActionItem : NSObject
+
+@property (nonatomic) NSString *title;
+@property (nonatomic) UIImage *icon;
+@property (nonatomic) OAActionButtonOnClick onClickFunction;
+
+- (instancetype)initWithIcon:(UIImage *)icon title:(NSString *)title onClickFunction:(OAActionButtonOnClick)onClickFunction;
+- (void)onClick;
+
+@end
+
+@implementation OAActionItem
+
+- (instancetype)initWithIcon:(UIImage *)icon title:(NSString *)title onClickFunction:(OAActionButtonOnClick)onClickFunction
+{
+    self = [super init];
+    if (self) {
+        _title = title;
+        _icon = icon;
+        _onClickFunction = onClickFunction;
+    }
+    return self;
+}
+
+- (void)onClick
+{
+    self.onClickFunction(self);
+}
+
+@end
+
+@interface OARearrangeCustomFiltersViewController() <UITableViewDelegate, UITableViewDataSource>
+
+@property (weak, nonatomic) IBOutlet UIView *navBar;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
+@property (weak, nonatomic) IBOutlet UIButton *cancelButton;
+@property (weak, nonatomic) IBOutlet UIButton *doneButton;
+
+@end
+
+@implementation OARearrangeCustomFiltersViewController
+{
+    OAAppSettings *_settings;
+    OAPOIFiltersHelper *_filtersHelper;
+    BOOL _isChanged;
+    BOOL _orderModified;
+    BOOL _hiddenModified;
+    BOOL _wasReset;
+
+    NSMutableArray<OAActionItem *> *_actionsItems;
+    NSMutableArray<OAEditFilterItem *> *_filtersItems;
+    NSMutableArray<OAEditFilterItem *> *_hiddenFiltersItems;
+    NSMapTable<NSString *, NSNumber *> *_filtersOrders;
+    NSMutableArray<NSString *> *_hiddenFiltersKeys;
+}
+
+- (instancetype)initWithFilters:(NSArray<OAPOIUIFilter *> *)filters
+{
+    self = [super init];
+    if (self)
+    {
+        _settings = [OAAppSettings sharedManager];
+        _filtersHelper = [OAPOIFiltersHelper sharedInstance];
+        _orderModified = [_settings poiFiltersOrder] != nil;
+        _hiddenModified = [_settings inactivePoiFilters] != nil;
+        [self generateData:filters];
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    [self.tableView setEditing:YES];
+}
+
+- (void)applyLocalization
+{
+    self.titleLabel.text = OALocalizedString(@"rearrange_categories");
+    [self.cancelButton setTitle:OALocalizedString(@"shared_string_cancel") forState:UIControlStateNormal];
+    [self.doneButton setTitle:OALocalizedString(@"shared_string_done") forState:UIControlStateNormal];
+}
+
+- (UIView *)getTopView
+{
+    return _navBar;
```

Comment:
Do not use these obsolete methods, they were used for correct view controller layout before constraints.